notificationworkerlambda/src/main/scala/com/gu/notifications/worker/delivery/fcm/FcmClient.scala

```diff
@@ -80,6 +104,24 @@ class FcmClient (firebaseMessaging: FirebaseMessaging, firebaseApp: FirebaseApp,
       Left(UnknownReasonFailedRequest(notificationId, token))
   })
 
+  def parseBatchSendResponse(
+    notificationId: UUID, tokens: List[String], triedResponse: Try[BatchResponse]
+  )(cb: Either[Throwable, Success] => Unit): Unit = triedResponse match {
+    case Success(batchResponse) =>
+      // From firebase sdk docs: the order of the response list corresponds to the order of the input tokens
+      batchResponse.getResponses.asScala.toList.zip(tokens).foreach { el => {
```

Comment:
We need to associate any failure response to the corresponding token. The batch response alone doesn't facilitate this, but the docs indicate that the order of the batch responds corresponds to the order of the input.